Documentation/development-flow.md

```diff
@@ -52,6 +52,65 @@ git remote -v
 ```
 Now you should have at least `origin` and `upstream` remotes. You can also add other remotes to collaborate with other contributors.
 
+## Layout
+
+A source code layout is shown below, annotated with comments about the use of each important directory:
+
+rook  
+├── build  
+├── cluster  
+│   ├── charts                    # Helm charts  
+│   │   └── rook-ceph  
+│   └── examples                  # Sample yaml files for rook cluster  
+│  
+├── cmd                           # Binaries with main entrypoint  
+│   ├── rook                      # Main command entry points for operators and daemons  
+│   └── rookflex                  # Main command entry points for rook flexvolume driver  
+│  
+├── design                        # Design documents for the various components of the rook project  
+├── Documentation                 # Rook project Documentation  
+├── images                        # Dockerfiles for images  
+│  
+├── pkg  
+│   ├── apis  
+│   │   ├── ceph.rook.io          # ceph specific specs for cluster, file, object  
+│   │   │   ├── v1alpha1  
+│   │   │   └── v1beta1  
+│   │   ├── cockroachdb.rook.io   # cockroachdb specific specs  
+│   │   │   └── v1alpha1  
+│   │   ├── minio.rook.io         # minio specific specs for cluster, object  
+│   │   │   └── v1alpha1  
+│   │   ├── nfs.rook.io           # nfs server specific specs  
+│   │   │   └── v1alpha1  
+│   │   └── rook.io               # rook.io API group of common types  
+│   │       ├── v1alpha1  
+│   │       └── v1alpha2  
+│   ├── client                    # auto-generated strongly typed client code to access Rook APIs  
+│   ├── clusterd  
+│   ├── daemon                    # daemons for each storage backend  
+│   │   ├── ceph  
+│   │   └── discover  
+│   ├── operator                  # all orchestration logic and custom controllers for each storage backend  
+│   │   ├── ceph  
+│   │   ├── cockroachdb  
+│   │   ├── discover  
+│   │   ├── k8sutil  
+│   │   ├── minio  
+│   │   ├── nfs  
+│   │   └── test  
+│   ├── test  
+│   ├── util  
+│   └── version  
+├── tests                         # integration tests  
+│   ├── framework                 # the rook testing framework  
+│   │   ├── clients               # consume rook resources  
```

Comment:
these descriptions for clients and installer are a bit too short, perhaps they should also mention that they are only for testing contexts? e.g.,
```
clients       # test clients used to consume Rook resources during integration tests
installer    # installs Rook and its supported storage providers into integration tests environments
```